nil/internal/config/params.go

```diff
@@ -152,40 +152,65 @@ func mergeValidators(input []ListValidators) []ValidatorInfo {
 	return result
 }
 
-func GetValidatorListForShard(
-	ctx context.Context, database db.DB, height types.BlockNumber, shardId types.ShardId, logger zerolog.Logger,
-) ([]ValidatorInfo, error) {
+func GetCurrentConfigAccessor(ctx context.Context, database db.DB, block *types.Block, shardId types.ShardId) (ConfigAccessor, error) {
 	tx, err := database.CreateRoTx(ctx)
 	if err != nil {
 		return nil, err
 	}
 	defer tx.Rollback()
+	return GetCurrentConfigAccessorTx(tx, block, shardId)
+}
+
+func GetCurrentConfigAccessorTx(tx db.RoTx, block *types.Block, shardId types.ShardId) (ConfigAccessor, error) {
+	var mainShardHash *common.Hash
+	if block != nil {
+		mainShardHash = &block.MainChainHash
+		if shardId.IsMainShard() {
+			mainShardHash = &block.PrevBlock
+			// The first block uses configuration from itself.
+			if mainShardHash.Empty() {
+				h := block.Hash(types.MainShardId)
```

Comment:
Can you please explain what happens here?

Reply:
I'm not sure I understand either. Didn't you already pass "prev block" as `block` parameter? What is the purpose of calling another `&block.PrevBlock` here?

Reply:
Configuration of the first block should be calculated on this block itself. Since it's the first block and you don't have any previous committed.

Reply:
What is the purpose of calling another &block.PrevBlock here

because mainShard has MainChainHash empty for itself. And we need to choose `block.PrevBlock`. Just for consistency.